### ISSUE TITLE
A bug in NEFullyConnectedLayer output size?

### ISSUE THREAD

**Issue by liviolima80** [ARM-software/ComputeLibrary#147]
Hi all,
I have an issue related to the NEFullyConnectedLayer output size. I have the following code

```cpp
Tensor input;
Tensor out_hidden, weights, bias;
input = create_tensor_1D(  in_size );
weights = create_tensor_2D( out_size, in_size );
bias = create_tensor_1D( out_size );
out_hidden = create_tensor_1D( out_size );
fully.configure(&input, &weights, &bias, &out_hidden, false);
```

where
```cpp
Tensor create_tensor_1D(int x )
{
	Tensor tensor;
	const TensorShape shape(x);
	tensor.allocator()->init(TensorInfo(shape, 1, DataType::F32));
	tensor.allocator()->allocate();
	return tensor;
}

Tensor create_tensor_2D(int x, int y )
{
	Tensor tensor;
	const TensorShape shape(x, y);
	tensor.allocator()->init(TensorInfo(shape, 1, DataType::F32));
	tensor.allocator()->allocate();
	return tensor;
}
```

As value of in_size seems that i can use every value, but as value of out_size seems that there are limitations. Seems that the minimum acceptable value is 16. If I use for example 10 I get the following error on configure method

```
terminate called after throwing an instance of 'std::runtime_error'
  what():  in validate ./arm_compute/core/Window.inl:73: _dims[i].end() <= _dims[i].start()
Aborted (core dumped)
```

Thanks

**Comment by AnthonyARM** [comment]
You need to call the function configure before your tensor allocate:
```cpp
tensor.allocator()->init() // Initialize the tensor dimensions
function.configure( tensor ) // Update the padding requirements of the tensor
tensor.allocator()->allocate() // Allocate the memory
```
If you call the allocate() before the configure then 
```cpp
tensor.allocator()->init() // Initialize the tensor dimensions
tensor.allocator()->allocate() // Allocate the memory
function.configure( tensor ) // Padding can't grow, therefore the execution window will be shrunk
```
The problem is that you're using a small tensor, therefore when the execution window is shrunk it becomes a window of size 0 which is what triggers the assert.

**Comment by liviolima80** [comment]
Thank you,
you are right, I changed too much the examples... ;-) Now everything works fine.

I just finish the implementation of a MLP (input -> fully -> relu -> fully -> relu -> fully -> softmax) network for NMINST classification loading weights from model training in Keras and everything works perfecty.

Thank you so much for this library!

**Comment by AnthonyARM** [comment]
Cool, glad it works :)
